test/javascripts/components/chat-channel-preview-card-test.js

```diff
@@ -0,0 +1,87 @@
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
```

Comment:
we should have a test for onJoinChannel

Reply:
I am not sure the best way to test this for a component test? The ember guide mentions using service stubs which we do have for chat https://guides.emberjs.com/release/testing/testing-components/ but I'm not sure what exactly I should be testing wrt AJAX calls and things like that. The chat-browse-test already checks that the join button works and the card is gone / channel is reloaded. I am going to merge this PR, but would like your opinion on this and I can add another test afterward if needed.

Reply:
I think what we want to test here os that clicking the button calls an action named onJoinChannel

Reply:
There doesn't really seem to be an easy way to do that though? In https://guides.emberjs.com/release/testing/testing-components/#toc_testing-actions they presume that the action is being passed in via args like this:

```
export default class CommentFormComponent extends Component {
  @tracked comment = '';

  @action
  submitComment() {
    this.args.submitComment({ comment: this.comment });
  }
}
```

```
<CommentForm @submitComment={{this.externalAction}} />
```

Which is not the case for the preview card. I've tried a few different ways of overriding `onJoinChannel` in the test with a stub and none of them worked. And anyway not sure what value testing it here adds when we already test clicking the button in chat-browse-test? If you had an example of what you mean that would be great.

Reply:
I looked at this today and made it work but I want to write more dedicated guidelines for this. Im just like you and I'm learning to adapt to our new patterns and also improve what we currently do.

I think the good separation is:
- one happy path acceptance test
- lots of component/unit tests for various edge cases

I will elaborate more on this in the near future.

Reply:
Nice, thank you for looking into this!